libraries/AP_HAL/board/chibios.h

```diff
@@ -179,3 +179,10 @@
 #ifndef AP_CPU_IDLE_STATS_ENABLED
 #define AP_CPU_IDLE_STATS_ENABLED HAL_PROGRAM_SIZE_LIMIT_KB > 1024
 #endif
+
+#if defined(STM32H7) && HAL_MEM_CLASS >= HAL_MEM_CLASS_1000
+// 32k gives huge performance improvements on boards that can cope
+// the memory check excludes things like STM32H750 and STM32H730
+#define AP_FATFS_MAX_IO_SIZE 32768
+#define AP_FATFS_MIN_IO_SIZE 4096
```

Comment:
These should also be `#ifndef`-wrapped so hwdefs can override them.

Reply:
Thanks fixed